### ISSUE TITLE
[BUG] rspamd with antivirus clamav fails on macOS with "IO read error: unexpected EOF"

### ISSUE THREAD

**Issue by essandess** [rspamd/rspamd#2879]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

Perhaps related to these (unresolved) issues:
* https://github.com/rspamd/rspamd/issues/479
* https://github.com/rspamd/rspamd/issues/1866
* https://github.com/rspamd/rspamd/issues/597

**Describe the bug**

I'm very much enjoying my [brand new](https://github.com/macports/macports-ports/pull/4201) `rspamd` install. But I see an issue when trying to use `clamav` with antivirus.

**Without** the `local.d/antivirus.conf` file, `rspamc` works great:

```
rspamc ~/Downloads/Test.eml 
Results for file: /Users/admin/Downloads/Test.eml (0.722 seconds)
[Metric: default]
Action: no action
Spam: false
Score: 1.19 / 15.00
Symbol: ARC_ALLOW (-1.00)[i=1]
…
```

But **with** a simple `local.d/antivirus.conf` file, `rspamc` fails after a restart:

First, kill and relaunch an `rspamd` process:
```
sudo port unload rspamd ; sudo port load rspamd
--->  Unloading startupitem 'rspamd' for rspamd
--->  Loading startupitem 'rspamd' for rspamd
```

Run `rspamc`:

```
rspamc ~/Downloads/Test.eml 
Results for file: /Users/lcladmin/Downloads/Test.eml (0.046 seconds)
IO read error: unexpected EOF

```

Check http://localhost:11334/#history, see no errors.

I know that my `clamd`'s `localhost instance and socket are not rock solid:

```
clamdscan ~/Downloads/Test.eml 
/Users/admin/Downloads/Test.eml: OK

----------- SCAN SUMMARY -----------
Infected files: 0
Time: 0.001 sec (0 m 0 s)
```

The key difference is setting `servers` in `local.d/antivirus.conf`. This fail the same way whether the socket or localhost is used.

```
clamav {
  type = "clamav";
  servers = "127.0.0.1:3310";
  # servers = "/opt/local/var/run/clamav/clamd.socket";
  patterns {
    # symbol_name = "pattern";
    JUST_EICAR = '^Eicar-Test-Signature$';
  }
  whitelist = "/opt/local/etc/rspamd/antivirus.wl";
}
```

This failure affects the entire `rspamd` functionality.

### Steps to Reproduce

Set `clamav { servers = "127.0.0.1:3310"; }` in `local.d/antivirus.conf`.

**Expected behavior**

`rspamc` doesn't break.

### Versions

```
rspamd --version
Rspamd daemon version 1.9.2

clamd --version
ClamAV 0.101.2/25436/Wed May  1 03:58:19 2019

echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
macOS 10.14.4 18E226
Xcode 10.2.1 10E1001
```

### Additional Information

In `local.d/logging.inc`, I have:
```
level = "debug";
debug_modules = ["redis", "antivirus"];
```

Also ,`rspamd` is awesome.

**Comment by essandess** [comment]
Follow-up: when I do see errors in the controller, they look like this. Those pids are all children processes that `ps -ef | grep rspamd` says look like this:
```
/opt/local/bin/rspamd -u _rspamd -g _rspamd -f -c /opt/local/etc/rspamd/rspamd.conf
```

The errors in the web page look like:
```
caught fatal signal 11(Segmentation fault: 11), pid: 28644, trace:
caught fatal signal 11(Segmentation fault: 11), pid: 28643, trace:
caught fatal signal 11(Segmentation fault: 11), pid: 28473, trace:
…
cannot load rrd from /opt/local/var/lib/rspamd/rspamd.rrd: rrd file seems to have incorrect size: 797696, must be -689968
caught fatal signal 11(Segmentation fault: 11), pid: 28032, trace:
```

After seeing the `.rrd` error, I did shut down the process, `rm /opt/local/var/lib/rspamd/*` and relaunched, but the issues persist (without that `.rrd` error).

The (perhaps) corresponding entries in `rspamd.log` look like:
```
2019-05-01 12:06:05 #23975(controller) rspamd_crash_sig_handler: caught fatal sign
al 11(Segmentation fault: 11), pid: 23975, trace: 
2019-05-01 12:06:05 #23940(main) <92b30e>; main; rspamd_cld_handler: catch SIGCHLD
 signal, finding terminated workers
2019-05-01 12:06:05 #23940(main) <92b30e>; main; rspamd_cld_handler: controller process 23975 terminated abnormally by signal: Segmentation fault: 11 but NOT created core file (throttled=no); core file limits: 0 current, 9223372036854775807 max
…
2019-05-01 12:16:04 #23977(normal) rspamd_worker_on_terminate: performing finishin
g actions
2019-05-01 12:16:04 #23977(normal) rspamd_crash_sig_handler: caught fatal signal 11(Segmentation fault: 11), pid: 23977, trace: 
2019-05-01 12:16:04 #23974(rspamd_proxy) rspamd_crash_sig_handler: caught fatal signal 11(Segmentation fault: 11), pid: 23974, trace: 
2019-05-01 12:16:04 #23979(normal) rspamd_crash_sig_handler: caught fatal signal 11(Segmentation fault: 11), pid: 23979, trace: 
2019-05-01 12:16:04 #23978(normal) rspamd_crash_sig_handler: caught fatal signal 11(Segmentation fault: 11), pid: 23978, trace: 
2019-05-01 12:16:04 #23976(normal) rspamd_crash_sig_handler: caught fatal signal 11(Segmentation fault: 11), pid: 23976, trace: 
2019-05-01 12:16:04 #23981(controller) rspamd_crash_sig_handler: caught fatal signal 11(Segmentation fault: 11), pid: 23981, trace: 
```

**Comment by essandess** [comment]
I'm closing this because I see a bad configuration error: I discovered that I had colliding sockets with 
```
control_socket = "$DBDIR/rspamd.sock mode=0600";
```
in `options.inc` **and** 
```
bind_socket = "$DBDIR/rspamd.sock mode=0666 owner=_rspamd";
```
in `local.d/worker-controller.inc`. Oops.

I changed the `control` worker socket to
```
bind_socket = "$RUNDIR/rspamd.sock mode=0666 owner=_rspamd";
```
and all the Segmentation faults went away.

Unfortunately, I'm still seeing the `IO read error: unexpected EOF`, independent of `clamav`.

I strongly suspect this is also caused from some misconfiguration, I just don't know what yet. 

I'm resting to factory and trying to track it down:
```
sudo rm /opt/local/var/lib/rspamd/*
redis-cli FLUSHALL
```
